------------------------------------------------------------------------
r84950 | johnmaddock | 2013-07-04 02:13:23 -0700 (Thu, 04 Jul 2013) | 2 lines

Patch for recent versions of glibc which always assume int64_t support.
Fixes #8731.

Upstream-Status: Backport

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: boost_1_54_0/boost/cstdint.hpp
===================================================================
--- boost_1_54_0.orig/boost/cstdint.hpp
+++ boost_1_54_0/boost/cstdint.hpp
@@ -41,7 +41,10 @@
 // so we disable use of stdint.h when GLIBC does not define __GLIBC_HAVE_LONG_LONG.
 // See https://svn.boost.org/trac/boost/ticket/3548 and http://sources.redhat.com/bugzilla/show_bug.cgi?id=10990
 //
-#if defined(BOOST_HAS_STDINT_H) && (!defined(__GLIBC__) || defined(__GLIBC_HAVE_LONG_LONG))
+#if defined(BOOST_HAS_STDINT_H)					\
+  && (!defined(__GLIBC__)					\
+      || defined(__GLIBC_HAVE_LONG_LONG)			\
+      || (defined(__GLIBC__) && ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 17)))))
 
 // The following #include is an implementation artifact; not part of interface.
 # ifdef __hpux
